include/gatekeeper_log_ratelimit.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Gatekeeper - DoS protection system.
+ * Copyright (C) 2016 Digirati LTDA.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _GATEKEEPER_LOG_RATELIMIT_H_
+#define _GATEKEEPER_LOG_RATELIMIT_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <rte_memory.h>
+
+/*
+ * The code of this file is mostly a copy of the Linux kernel.
+ */
+
+struct log_ratelimit_state {
+	uint32_t ratelimit_level;
+	uint64_t interval_cycles;
```

Comment:
Once `log_ratelimit_allow()` become `static` (see another comment), we can move `struct log_ratelimit_state` to `lib/log_ratelimit.c`.